lectures/numpy.md

```diff
@@ -475,6 +478,456 @@ A @ (0, 1)
 
 Since we are post-multiplying, the tuple is treated as a column vector.
 
+(broadcasting)=
+## Broadcasting
+
+```{index} single: NumPy; Broadcasting
+```
+
+(This section is built upon an excellent discussion of broadcasting provided by [Jake VanderPlas](https://jakevdp.github.io/PythonDataScienceHandbook/02.05-computation-on-arrays-broadcasting.html).)
+
+```{note}
+Some aspects of broadcasting are hard to visualize and considered relatively advanced. As such this section can be skimmed at first pass.
+```
```

Comment:
On second thoughts, let's change the wording of this note to:

Broadcasting is a very important aspect of NumPy.  At the same time, advanced broadcasting is relatively complex and some of the details below can be skimmed on first pass.